tests/js/set-tests.js

```diff
@@ -0,0 +1,392 @@
+
+        TestCase.assertEqual(false, java_team.Names.has("Kenneth"), "Kenneth shouldn't be in the Java team");
+        TestCase.assertEqual(true, java_team.Names.has("Claus"), "Claus should be in the Java team");
+
+        //
+        // Set.clear() functionality
+        //
+
+        // remove everyone from the JS team
+        realm.write(() => {
+            js_team.Names.clear();
+        });
+
+        teams = realm.objects(people_schema.name);
+        TestCase.assertEqual(2, teams.length, "There should be two teams");
+        js_team = teams[0];
+        java_team = teams[1];
+        TestCase.assertEqual(0, js_team.Names.length, "There should be no people in the JS team");
+        TestCase.assertEqual(2, java_team.Names.length, "There should be 2 people in the Java team");
+
+        realm.close();
+    },
+
+    // TODO:  Move Sync tests to separate file
+    async testSetSyncedNonRequired() {
+        // test that we can create a synced realm with a Set
+        // that isn't required
+        if (!global.enableSyncTests) return;
+
+        const schema = {
+            name: "SyncedSetInt",
+            primaryKey: "_id",
+            properties: {
+                _id: "int",
+                intSet: "int<>",
+            }
+        }
+
+        const appConfig = AppConfig.integrationAppConfig;
+        const app = new Realm.App(appConfig);
+        const credentials = Realm.Credentials.anonymous();
+
+        const user = await app.logIn(credentials);
+        const config = {
+            sync: {
+                user,
+                partitionValue: "_id",
+                _sessionStopPolicy: "immediately", // Make it safe to delete files after realm.close()
+            },
+            schema: [schema]
+        };
+        const realm = await Realm.open(config);
+
+
+        realm.write(() => {
+            realm.deleteAll();
+        });
+
+        let objects = realm.objects(schema.name);
+        let objcount = objects.length;
+
+        TestCase.assertEqual(0, objcount, "Table should be empty");
+    },
+
+    async testSetSyncedAddDelete() {
+        // tests a synced realm while adding/deleting elements in a Set
+        if (!global.enableSyncTests) return;
+
+        const schema = {
+            name: "SyncedSetInt",
+            primaryKey: "_id",
+            properties: {
+                _id: "int",
+                intSet: "int<>",
+            }
+        }
+        
+        const appConfig = AppConfig.integrationAppConfig;
+        const app = new Realm.App(appConfig);
+//        Realm.App.Sync.setLogLevel(app, 'all');
+        const credentials = Realm.Credentials.anonymous();
+
+        const user = await app.logIn(credentials);
+        const config = {
+            sync: {
+                user,
+                partitionValue: "_id",
+                _sessionStopPolicy: "immediately", // Make it safe to delete files after realm.close()
```

Comment:
Same as above - we should probably register a logger to ensure no sync related errors are logged.

Reply:
Scrap that. We should upload, delete the client-side Realm file and download the changes from the server to ensure we can roundtrip correctly.